content/blog/living-off-the-ipad-as-an-engineer/index.md

```diff
@@ -0,0 +1,110 @@
+
+![A quite simple desk setup. Doing some LumaFusion edits!](setup.jpeg)
+
+*Update: After quite some time with Magic Keyboard for iPad on the desk, I revised the setup, with the tablet laying flat under the monitor to quickly take handwritten notes or drawings, with the Keychron K2 + Magic Trackpad 2 as input devices. We’ll see if it sticks!*
+
+## Software Dev: Remote Server and Native Apps
+
+Let’s get into real engineering tools. As you may know very well already, there’s no walled garden like iOS/iPadOS. Apps are fully contained, *Files* is some kind of explorer yet remains very limited, and firing a local command line is pure fantasy.
+
+*How the heck do we write and run our beloved code, Pierre? 💁*
+
+As it’s common these days, the trick is in the cloud. While it comes with its drawbacks such as with spotty connections, offloading the work to a remote, safe, always-accessible machine has some nice things going for it. A hosted instance can be fired in minutes these days. To stay somewhat minimalist, I chose to keep an old laptop plugged in a closet, but there are many options out there.
+
+There’s essentially two solutions that work for me at the moment: [Visual Studio Code](https://code.visualstudio.com/) in the browser, which can be achieved in various ways; or a native SSH client app such as [Termius](https://apps.apple.com/us/app/termius-ssh-client/id549039908) or [Blink Shell](https://blink.sh/). The former is a cross-platform cloud SSH/Mosh service that has really good iPad support, its main downside being the lack of a full external monitor integration, as mentioned earlier. Blink Shell features it, and it works quite well. Picture that clean tmux/vim session on a 4K monitor, at the cost of losing mouse support, though.
+
+![A mobile web dev session in the native app Termius, with the side-kick browser](termius.png)
+
+For a proper IDE, there are ways to run the undisputed leader of the last years—VS Code — in Safari. I personally only fire it for refactoring and large Git diffs, I’m happy with terminal-only work otherwise. [code-server](http://code-server.dev/) is a project that does exactly that, and at the time of writing comes with a Beta parameter --link [that does all the hard work of securing the connection](https://github.com/cdr/code-server#cloud-program-%EF%B8%8F). One line to install on various hosts, and one port-forwarding command to start the remote access. For more details on the setup with more technical bits, you can check out [my Notion project page](https://www.notion.so/iPad-Challenge-ba216d5956194453a0dd4d56f62d888c).
```

Comment:
For a proper IDE, there are ways to run the undisputed leader of the last years — VS Code  — in Safari. I personally only fire it up for refactoring and large Git diffs. I’m happy with terminal-only work otherwise. [Code-server](http://code-server.dev/) is a project that does exactly that, and at the time of writing comes with a Beta parameter --link [that does all the hard work of securing the connection](https://github.com/cdr/code-server#cloud-program-%EF%B8%8F). A one line command is all it takes to install on various hosts, and another one-liner for port-forwarding to start the remote access. For more details on the setup with more technical bits, you can check out [my Notion project page](https://www.notion.so/iPad-Challenge-ba216d5956194453a0dd4d56f62d888c).

Reply:
Waiting on the em-dash “discussion”, I love the other changes.